src/animation/AnimationMixer.js

```diff
@@ -199,42 +315,139 @@ THREE.AnimationMixer.prototype = {
 
 	update: function( deltaTime ) {
 
-		var mixerDeltaTime = deltaTime * this.timeScale;
-		this.time += mixerDeltaTime;
+		var actions = this._actions,
+			mixerDeltaTime = deltaTime * this.timeScale;
 
-		for ( var i = 0; i < this.actions.length; i ++ ) {
+		var time = this.time += mixerDeltaTime;
+		var accuIndex = this.accuIndex ^= 1;
 
-			var action = this.actions[i];
+		// perform all actions
 
-			var weight = action.getWeightAt( this.time );
+		for ( var i = 0, n = actions.length; i !== n; ++ i ) {
 
-			var actionTimeScale = action.getTimeScaleAt( this.time );
-			var actionDeltaTime = mixerDeltaTime * actionTimeScale;
+			var action = actions[ i ];
+			if ( ! action.enabled ) continue;
 
-			var actionResults = action.update( actionDeltaTime );
+			var weight = action.getWeightAt( time );
+			if ( weight <= 0 ) continue;
 
-			if ( action.weight <= 0 || ! action.enabled ) continue;
+			var actionTimeScale = action.getTimeScaleAt( time );
+			var actionTime = action.updateTime( mixerDeltaTime * actionTimeScale );
 
-			for ( var j = 0; j < actionResults.length; j ++ ) {
+			var interpolants = action._interpolants;
+			var propertyMixers = action._propertyBindings;
 
-				var name = action.clip.tracks[j].name;
+			for ( var j = 0, m = interpolants.length; j !== m; ++ j ) {
 
-				action.propertyBindings[ j ].accumulate( actionResults[j], weight );
+				interpolants[ j ].getAt( actionTime );
```

Comment:
The constructor of `Action` calls `track.createInterpolant( null )`, where `null` is the `.result` that is attached later in `mixer.addAction`. The track type decides which exact interpolant to use.

The interpolant's `.result` is assigned to `propertyMixer.buffer`. Was missing a reset to `null` in `mixer.removeAction` (for historic reasons, I used to cache stuff on the actions - but got rid of it), now fixed.